### PR TITLE
massive training time improvement

### PR DESCRIPTION
**Pull request by IgnacioJPickering**
Massive training time improvement by doing a simple substitution in nnp_training

### CODE REVIEW

**Comment by farhadrgh** [comment]
I think the issue is shuffled dataset is passing chunks in CPU device. You will need to:
```
chunk_species = chunk_species.to(parser.device)
chunk_coordinates = chunk_coordinates.to(parser.device)
```

**Comment by IgnacioJPickering** [comment]
@farhadrgh yes, you are right.

File: tools/training-benchmark.py [hunk 1]
```diff
@@ -163,16 +163,17 @@ def wrapper(*args, **kwargs):
...
             predicted_energies = []
             num_atoms = []
+            atomic_properties = []
 
             for chunk_species, chunk_coordinates in batch_x:
-                chunk_species = chunk_species.to(parser.device)
-                chunk_coordinates = chunk_coordinates.to(parser.device)
+                atomic_chunk = {'species': chunk_species.to(parser.device), 'coordinates': chunk_coordinates.to(parser.device)}
+                atomic_properties.append(atomic_chunk)
                 num_atoms.append((chunk_species >= 0).to(true_energies.dtype).sum(dim=1))
```
Comment (farhadrgh) [inline tools/training-benchmark.py]: `num_atoms` is still on CPU. Maybe we should do:
```
chunk_species = chunk_species.to(parser.device)
chunk_coordinates = chunk_coordinates.to(parser.device)
```
and then use the updated `chunk_species`?

Comment (IgnacioJPickering) [reply tools/training-benchmark.py]: Ahh, yes, will do

Comment (farhadrgh) [reply tools/training-benchmark.py]: great, thanks, I am going to add a couple of notes here then merge

Comment (IgnacioJPickering) [reply tools/training-benchmark.py]: Hey, why aren't tests running?

Comment (IgnacioJPickering) [reply tools/training-benchmark.py]: fixed I guess

**Comment by farhadrgh** [comment]
Here are some benchmarking results on GeForce RTX 2080, Intel i9-9900K using %100 of ANI1x dataset with `batch_size` of 2560:

### 1. A comparison between different APIs before this PR:
**- Old data API**
```
torchani.aev.cutoff_cosine - 3.9s
torchani.aev.radial_terms - 3.9s
torchani.aev.angular_terms - 11.6s
torchani.aev.compute_shifts - 0.0s
torchani.aev.neighbor_pairs - 0.0s
torchani.aev.neighbor_pairs_nopbc - 7.0s
torchani.aev.triu_index - 0.0s
torchani.aev.cumsum_from_zero - 2.1s
torchani.aev.triple_by_molecule - 23.0s
torchani.aev.compute_aev - 61.6s
Total AEV - 61.9s
Data Loading - 38.2s
NN - 44.4s
Epoch time - 151.3s
```
**- New data API (Cached)**
```
torchani.aev.cutoff_cosine - 0.5s
torchani.aev.radial_terms - 0.5s
torchani.aev.angular_terms - 1.6s
torchani.aev.compute_shifts - 0.0s
torchani.aev.neighbor_pairs - 0.0s
torchani.aev.neighbor_pairs_nopbc - 1.3s
torchani.aev.triu_index - 0.0s
torchani.aev.cumsum_from_zero - 1.2s
torchani.aev.triple_by_molecule - 6.2s
torchani.aev.compute_aev - 13.2s
Total AEV - 13.2s
Data Loading - 22.7s
NN - 25.9s
Epoch time - 51.5s
```
**- New data API (Shuffled)**
```
torchani.aev.cutoff_cosine - 0.6s
torchani.aev.radial_terms - 0.6s
torchani.aev.angular_terms - 1.7s
torchani.aev.compute_shifts - 0.0s
torchani.aev.neighbor_pairs - 0.0s
torchani.aev.neighbor_pairs_nopbc - 1.4s
torchani.aev.triu_index - 0.0s
torchani.aev.cumsum_from_zero - 1.2s
torchani.aev.triple_by_molecule - 6.5s
torchani.aev.compute_aev - 13.9s
Total AEV - 13.9s
Data Loading - 24.6s
NN - 25.9s
Epoch time - 78.5s
```

### 2. A comparison between different APIs after merging this PR:
**- Old data API**
```
torchani.aev.cutoff_cosine - 0.3s
torchani.aev.radial_terms - 0.3s
torchani.aev.angular_terms - 0.8s
torchani.aev.compute_shifts - 0.0s
torchani.aev.neighbor_pairs - 0.0s
torchani.aev.neighbor_pairs_nopbc - 4.8s
torchani.aev.triu_index - 0.0s
torchani.aev.cumsum_from_zero - 1.6s
torchani.aev.triple_by_molecule - 6.2s
torchani.aev.compute_aev - 16.1s
Total AEV - 16.2s
Data Loading - 38.0s
NN - 28.0s
Epoch time - 58.7s
```
**- New data API (Cached)**
```
torchani.aev.cutoff_cosine - 0.3s
torchani.aev.radial_terms - 0.3s
torchani.aev.angular_terms - 0.8s
torchani.aev.compute_shifts - 0.0s
torchani.aev.neighbor_pairs - 0.0s
torchani.aev.neighbor_pairs_nopbc - 4.8s
torchani.aev.triu_index - 0.0s
torchani.aev.cumsum_from_zero - 1.6s
torchani.aev.triple_by_molecule - 6.2s
torchani.aev.compute_aev - 16.1s
Total AEV - 16.1s
Data Loading - 23.2s
NN - 27.8s
Epoch time - 54.4s
```
**- New data API (Shuffled)**
```
torchani.aev.cutoff_cosine - 0.3s
torchani.aev.radial_terms - 0.3s
torchani.aev.angular_terms - 0.9s
torchani.aev.compute_shifts - 0.0s
torchani.aev.neighbor_pairs - 0.0s
torchani.aev.neighbor_pairs_nopbc - 4.8s
torchani.aev.triu_index - 0.0s
torchani.aev.cumsum_from_zero - 1.6s
torchani.aev.triple_by_molecule - 6.4s
torchani.aev.compute_aev - 16.4s
Total AEV - 16.5s
Data Loading - 24.5s
NN - 27.8s
Epoch time - 79.8s
```

**Comment by zasdfgbnm** [comment]
@farhadrgh Thanks!

**Pull request merged**